advisor/src/test/kotlin/advisors/VulnerableCodeTest.kt

```diff
@@ -0,0 +1,312 @@
+
+    result shouldNotBeNull {
+        this.map { it.id } should containExactlyInAnyOrder(packageIdentifiers)
+
+        packageIdentifiers.forEach { pkg ->
+            val pkgResults = advisorResultsFor(this, pkg)
+            pkgResults shouldHaveSize 1
+            val pkgResult = pkgResults[0]
+            pkgResult.vulnerabilities should beEmpty()
+            pkgResult.summary.issues shouldHaveSize 1
+            val issue = pkgResult.summary.issues[0]
+            issue.severity shouldBe Severity.ERROR
+        }
+    }
+}
+
+/**
+ * Create a configuration for the [VulnerableCode] advisor that points to the local [wireMockServer].
+ */
+private fun createConfig(wireMockServer: WireMockServer): VulnerableCodeConfiguration {
+    val url = "http://localhost:${wireMockServer.port()}"
+    return VulnerableCodeConfiguration(url)
+}
+
+/**
+ * Create a test advisor instance that communicates with the local [wireMockServer].
+ */
+private fun createAdvisor(wireMockServer: WireMockServer): VulnerableCode =
+    VulnerableCode(ADVISOR_NAME, createConfig(wireMockServer))
+
+/**
+ * Return the test file with an analyzer result.
+ */
+private fun resultFile(): File = File(TEST_FILES_ROOT).resolve(TEST_RESULT_NAME)
+
+/**
+ * Generate the JSON body of the request for a list of [items] using the given [label]. This is used for both
+ * the request for packages and vulnerabilities.
+ */
+private fun generateListRequest(items: List<String>, label: String): String =
+    items.joinToString(prefix = "{ \"$label\": [", postfix = "] }") { "\"$it\"" }
+
+/**
+ * Generate a JSON string with details about a vulnerability with the given [id], [cveId], and [score].
+ */
+private fun vulnerabilityDetails(id: String, cveId: String, score: Float): String =
+    vulnerabilityDetailsTemplate.replace("<<id>>", id)
+        .replace("<<cve>>", cveId)
+        .replace("<<score>>", score.toString())
+
+/**
+ * Stub a GET request for the details of the vulnerability with the given [id]. The request returns a JSON
+ * containing the given [cveId] and [score] with the specified [statusCode].
+ */
+private fun stubVulnerability(id: String, cveId: String, score: Float, statusCode: Int = 200) {
+    stubFor(
+        get(urlPathEqualTo("/api/vulnerabilities/$id"))
+            .willReturn(
+                aResponse().withStatus(statusCode)
+                    .withBody(vulnerabilityDetails(id, cveId, score))
+            )
+    )
+}
+
+/**
+ * Extract the advisor results for the package with the given [id] from the set of [results].
+ */
+private fun advisorResultsFor(results: SortedSet<AdvisorResultContainer>, id: Identifier): List<AdvisorResult> =
+    results.find { it.id == id }?.results ?: throw IllegalArgumentException("No result found for package $id.")
+
+/**
+ * Obtain all vulnerabilities from the advisor results in this collection.
+ */
+private fun Collection<AdvisorResult>.vulnerabilities(): List<Vulnerability> =
+    this.flatMap { it.vulnerabilities }
```

Comment:
Nit: Omit the superfluous `this.`.

Reply:
And I personally have a slight tendency towards inlining this.

Reply:
inlined.